### ISSUE TITLE
Please check `revise-internals` branch

### ISSUE THREAD

**Issue by marcolarosa** [describo/crate-builder-component#88]
Hi @beepsoft 

I hope you've had a good break. If possible, can you please checkout the branch [revise-internals](https://github.com/describo/crate-builder-component/tree/revise-internals) and make sure it's working as expected.

This is a total rewrite of the crate handling code because I finally had the time, and headspace over the break to rework it sensibly and remove a whole bunch of legacy code / ideas. This happened because I recently learnt about a way to expose the component internals to the outer app (`defineExpose`) which is useful to me and will likely be useful to you.

I've also spent a bunch of time bringing together all the describo related documentation at https://describo.github.io/documentation. The page https://describo.github.io/documentation/component/advanced-usage.html talks about what you can do with this new internal structure.

Anyway, if you can please take a look at the component in the dev app - it should work exactly as it did before.

And please have a look around the documentation. Specifically:
[ ] https://describo.github.io/documentation/team.html - are you happy to be on that page? What is your linkedin? Other social links - see : https://vitepress.dev/reference/default-theme-config.html#sociallinks

[ ] https://describo.github.io/documentation/component/advanced-usage.html
[ ] https://describo.github.io/documentation/component/crate-manager.html (API docs are in this branch so they're not yet published).
[ ] https://describo.github.io/documentation/component/themes.html

I still have some work to do in this branch but the major work is done.

Something to think about: Now that we can get to the internal crate handling code, we can expose the init method so that the outer application can deterministically load the crate and profile rather than depending on the vue prop handling which can fire at odd times. This isn't done yet but something like:

```
const describo = ref()

// load your crate and profile
// do any other setup things here
// then when ready:
describo.value.init({ crate, profile })

```

**Comment by marcolarosa** [comment]
Closing - released with https://github.com/describo/crate-builder-component/releases/tag/v0.56.0

**Comment by beepsoft** [comment]
Published 0.56.0 of the react component: https://www.npmjs.com/package/@describo/crate-builder-component-react
